src/main/java/org/terasology/web/resources/ConsoleResource.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2017 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.web.resources;
+
+import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.entitySystem.event.ReceiveEvent;
+import org.terasology.entitySystem.systems.RegisterSystem;
+import org.terasology.logic.console.Console;
+import org.terasology.logic.console.Message;
+import org.terasology.logic.console.MessageEvent;
+import org.terasology.network.ClientComponent;
+
+@RegisterSystem
+public class ConsoleResource extends EventEmittingResource<Message> implements DefaultComponentSystem, WritableResource<String> {
```

Comment:
Okay, I'm reading about the pattern and I'll keep it in mind for the future. However in this particular case inheritance fits better:
- In my understanding, since it needs to receive events, the class must be a `ComponentSystem` in some way; `DefaultComponentSystem` just extends `ComponentSystem` by providing do-nothing default implementations for the various preInitialize, initialize, etc methods.
- `WritableResource` provides a common interface for any resource that accepts write operations.
- `EventEmittingResource` works like `java.util.Observable`.

Maybe you already noticed but just in case, [here is the class diagram](https://gianluca-nitti.github.io/GSoC-2017-devlog/images/ResourceClassDiagram.PNG) for resource management if you are interested. I'm not sure how it could be refactored using composition, to me it looks okay since it's mostly a way to mark the various type of resources.